src/input/CommandQueue.java

```diff
@@ -0,0 +1,130 @@
+package input;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.LinkedList;
+
+import assets.Plant;
+import assets.PlantTypes;
+import assets.Zombie;
+import engine.Board;
+import engine.Game;
+import engine.Grid;
+import engine.Purse;
+import ui.GameUI;
+import util.Logger;
+
+/**
+ * This is the Command History Queue for the Player throughout a Game
+ * @author David Wang
+ */
+public class CommandQueue extends LinkedList<Command> implements Serializable {
+	private static final long serialVersionUID = 1L;
+	
+	private static Logger LOG = new Logger("Command Queue");
+	private Game game;
+	private GameUI ui;
+	private Board gameBoard;
+	private Purse userResources;
+	
+	public CommandQueue(Game game, GameUI ui, Board gB, Purse userResources) {
+		this.game = game;
+		this.ui = ui;
+		this.gameBoard = gB;
+		this.userResources = userResources;
+	}
+	
+	/**
+	 * Adds a Place Command to the Command History
+	 * @param type the planttype that was placed
+	 * @param x the location of the placement
+	 * @param y the location of the placement
+	 */
+	public void registerPlace(PlantTypes type, int x, int y) {
+		this.addFirst(new PlaceCommand(type, x, y));
```

Comment:
If the player tries to place a plant but he/she cannot afford it, this would still get stored in the stack of commands?